docs/src/components/GlobalSearch/index.js

```diff
@@ -0,0 +1,173 @@
+  return (
+    <div className="">
+      <Link to={`/${data.slug}`} className="search-result-link">
+        <div className="search-result-entry px-5 py-4">
+          <div className="d-flex align-items-center overflow-hidden">
+            <Text dangerouslySetInnerHTML={{ __html: getHighlightedText(data.frontmatter.title, query) }}></Text>
+            {
+              data.slug.includes('mobile') ?
+                <Icon className="ml-4" appearance='subtle' size={16} name='phone_iphone' /> :
+                <Icon className="ml-4" appearance='subtle' size={16} name='desktop_windows' />
+            }
+          </div>
+          <div>
+            <Text appearance='subtle' dangerouslySetInnerHTML={{ __html: getHighlightedText(data.frontmatter.description, query) }}></Text>
+          </div>
+        </div>
+      </Link>
+    </div>
+  )
+}
+
+const ShowResults = ({ name, list, query }) => {
+  return (
+    <div>
+      <Subheading className="pb-2 px-5 search-category" appearance="subtle">{name}</Subheading>
+      {list.map((data) => <CustomResultEntry query={query} data={data} />)}
+    </div>
+  )
+}
+
+const PageHit = ({ searchList, query }) => {
+
+  const components = [],
+    patterns = [],
+    foundations = [],
+    content = [],
+    resources = [],
+    introduction = [];
+
+  searchList.forEach((item) => {
+    const slugPath = item.slug.split('/');
+    let category = '';
+    if (slugPath[0] === 'mobile') {
+      category = slugPath[1];
+    } else {
+      category = slugPath[0];
+    }
+
+    switch (category) {
+      case 'components':
+        components.push(item);
+        break;
+
+      case 'patterns':
+        patterns.push(item);
+        break;
+
+      case 'foundations':
+        foundations.push(item);
+        break;
+
+      case 'content':
+        content.push(item);
+        break;
+
+      case 'resources':
+        resources.push(item);
+        break;
+
+      case 'introduction':
+        introduction.push(item);
+        break;
+
+      default:
+        break;
+    }
+  })
+
+  return (
+    <div>
+      {components.length > 0 && <ShowResults query={query} name="Components" list={components} />}
+      {patterns.length > 0 && <ShowResults query={query} name="Patterns" list={patterns} />}
+      {foundations.length > 0 && <ShowResults query={query} name="Foundations" list={foundations} />}
+      {content.length > 0 && <ShowResults query={query} name="Content" list={content} />}
+      {resources.length > 0 && <ShowResults query={query} name="Resources" list={resources} />}
+      {introduction.length > 0 && <ShowResults query={query} name="Introduction" list={introduction} />}
+    </div>
+  )
+}
+
+const Search = ({parentRef}) => {
+  const [query, setQuery] = useState(null);
+  const [searchResult, setSearchResult] = useState([]);
+  const [openPopover, setOpenPopover] = useState(false);
+  const searchList = useSearchItems();
+
+  // useEffect(() => {
```

Comment:
remove this if not needed or else add appropriate comments if we intend to use it in future